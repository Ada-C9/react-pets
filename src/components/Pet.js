import React, { Component } from 'react';
import PropTypes from 'prop-types';

const Pet = (props) => {

  return (
    <tr>
      <td>{props.name}</td>
      <td>{props.breed}</td>
      <td>{props.age}</td>
      <td>{props.about}</td>
    </tr>
  );
}

Pet.propTypes = {
  name: PropTypes.string.isRequired,
  breed: PropTypes.string.isRequired,
  age: PropTypes.number,
  about: PropTypes.string
};

export default Pet;
