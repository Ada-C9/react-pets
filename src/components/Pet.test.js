import React from 'react';
import { mount, shallow } from 'enzyme';

import Pet from './Pet';


describe('Pet', () => {
  test('that is matches an existing snapshot', () => {
    // Arrange
    // Mount the component in the DOM
    // Deep rendering
    const wrapper = shallow(<Pet
      name="Pete"
      breed="Pomeranian"
      age={23}
      about="cute!"
    />);

    // Assert
    expect(wrapper).toMatchSnapshot();
    // Remove the component from the DOM
    // Clean up
    wrapper.unmount();
  });
});