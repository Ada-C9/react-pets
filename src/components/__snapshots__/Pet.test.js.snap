// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pet that is matches an existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pet
    about="cute!"
    age={23}
    breed="Pomeranian"
    name="Pete"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Pet 
          Pete
</h3>,
        <p>
          <strong>
                    Age: 
          </strong>
          23
</p>,
        <p>
          <strong>
                    Breed: 
          </strong>
          Pomeranian
</p>,
        <p>
          cute!
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Pet ",
            "Pete",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Pet ",
          "Pete",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              Age: 
</strong>,
            23,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Age: ",
            },
            "ref": null,
            "rendered": "Age: ",
            "type": "strong",
          },
          23,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              Breed: 
</strong>,
            "Pomeranian",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Breed: ",
            },
            "ref": null,
            "rendered": "Breed: ",
            "type": "strong",
          },
          "Pomeranian",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "cute!",
        },
        "ref": null,
        "rendered": "cute!",
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Pet 
            Pete
</h3>,
          <p>
            <strong>
                        Age: 
            </strong>
            23
</p>,
          <p>
            <strong>
                        Breed: 
            </strong>
            Pomeranian
</p>,
          <p>
            cute!
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Pet ",
              "Pete",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Pet ",
            "Pete",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Age: 
</strong>,
              23,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Age: ",
              },
              "ref": null,
              "rendered": "Age: ",
              "type": "strong",
            },
            23,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Breed: 
</strong>,
              "Pomeranian",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Breed: ",
              },
              "ref": null,
              "rendered": "Breed: ",
              "type": "strong",
            },
            "Pomeranian",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "cute!",
          },
          "ref": null,
          "rendered": "cute!",
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
